### PR TITLE
Store active stack in local config

### PR DESCRIPTION
**Pull request by schustmi**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
- Store active stack in local config

### CODE REVIEW

File: src/zenml/core/base_component.py [hunk 1]
```diff
@@ -66,15 +69,37 @@ def __init__(self, **values: Any):
...
+        if self._superfluous_options:
+            logger.warning(
+                "Found superfluous configuration values for class `%s`: %s",
+                self.__class__.__name__,
+                set(self._superfluous_options),
+            )
+            config_path = self.get_serialization_full_path()
+            if os.path.exists(config_path):
+                backup_path = config_path + ".backup"
+                shutil.copyfile(config_path, backup_path)
```
Comment (htahir1) [inline src/zenml/core/base_component.py:85]: Does it make sense to just use path_utils here

**Review by htahir1 — COMMENTED** [review]
Can we add the following test to this branch: Create two global configs; one with normal options and one with superfluous options. Then write a few tests to see it works as expected?

File: src/zenml/core/base_component.py (hunk 1 above)
Comment (schustmi) [reply src/zenml/core/base_component.py:85]: Yep good idea

**Comment by schustmi** [comment]
They have a fixed file path, so we can't really test setting them in files I guess, but I can add tests where I pass in extra arguments to the init function which should also test this functionality

**Comment by htahir1** [comment]
You can mock / override the `get_serialization_dir` `get_serialization_file_name` methods using mocker or something to control that :-)

**Pull request merged**